File: MATTHEW/leetcode/AIcoding/5.Qwen-3/cpp_solutions/1525-number-of-good-ways-to-split-a-string.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

class Solution {
public:
    int numGoodPairs(string s) {
        vector<int> count(26, 0);
        int result = 0;
        for (char c : s) {
            count[c - 'a']++;
        }
        for (int i = 0; i < 26; i++) {
            if (count[i] >= 2) {
                result += count[i] * (count[i] - 1) / 2;
            }
        }
        return result;
    }
};

File: leetcode/AIcoding/4.GPT-4o/erlang_solutions/1525-number-of-good-ways-to-split-a-string.erl
-module(solution).
-export([num_good_ways/1]).

num_good_ways(S) ->
    Count = length(S),
    PrefixCounts = lists:foldl(fun(Char, {Count, Prefix}) ->
        NewPrefix = maps:get(Char, Prefix, 0) + 1,
        {Count, maps:put(Char, NewPrefix, Prefix)}
    end, {0, #{}}, S),
    maps:fold(fun(_, V, Acc) -> Acc + V end, 0, maps:map(fun(_, V) -> V * (Count - V) end, PrefixCounts)).